src/main/kotlin/no/nav/familie/ef/søknad/søknad/mapper/BarnMapper.kt

```diff
@@ -163,16 +163,33 @@ object BarnMapper : MapperMedVedlegg<List<Barn>, List<Kontraktbarn>>(BarnaDine)
         }
     }
 
+    private fun hvorforIkkeOppgiSvarIdBasertPåBegrunnelse(begrunnelse: String?): String? {
+        return when {
+            begrunnelse == "Donor" -> "donorbarn"
+            begrunnelse.erIkkeBlankEllerNull() -> "annet"
+            else -> null
+        }
+    }
+
     private fun String?.erIkkeBlankEllerNull() = !this.isNullOrBlank()
 
     private fun hvorforIkkeOppgiTilTekstfeltEllerNullBasertPåBegrunnelse(begrunnelse: String?): TekstFelt? {
         val hvorforIkkeOppgi = hvorforIkkeOppgiBasertPåBegrunnelse(begrunnelse)
-        if (hvorforIkkeOppgi != null) {
-            return TekstFelt("Hvorfor kan du ikke oppgi den andre forelderen?", hvorforIkkeOppgi)
+        val svarId = hvorforIkkeOppgiSvarIdBasertPåBegrunnelse(begrunnelse)
+        if (hvorforIkkeOppgi != null && svarId != null) {
+            return TekstFelt("Hvorfor kan du ikke oppgi den andre forelderen?", hvorforIkkeOppgi, svarId)
         }
         return null
     }
 
+    private fun søknadsfeltTilNullEllerNullableTekstFelt(søknadsFelt: Søknadsfelt<String>?): TekstFelt? {
+        return if (søknadsFelt?.verdi.isNullOrBlank() || søknadsFelt?.verdi == Språktekster.IkkeOppgitt.hentTekst()) {
```

Comment:
Her er det noe pussig. Navngivning av funksjonen gir ikke helt mening ift innhold.
Har vi satt locale på vei ut nå? `Språktekster.IkkeOppgitt.hentTekst()` krever jo at man har satt Locale.
Alternativ her er jo å sjekke om `hvorforIkkeOppgi` er satt - isåfall skal ikke navn settes

Reply:
Navnet var nok ikke optimalt. Slik jeg forstå det vil Språktekster.IkkeOppgitt.hentTekst() fungere fordi 'nb' er satt som default locale.

Har uansett gjort endringer til å sjekke om hvorforIkkeOppgi mangler verdi nå.